File: 2021/01/cpp/pt2.cpp
#include <fstream>
#include <iostream>
#include <limits>
#include <stdlib.h>
#include <vector>
using namespace std;

int main()
{
    // Read input
    vector<int> nums;
    fstream problemInput(realpath("../input.txt", NULL));
    int num = 0;
    while (problemInput >> num)
    {
        nums.push_back(num);
    }
    problemInput.close();

    int count;
    int prev = INT_MAX;

    for (int i = 2, l = nums.size(); i < l; i++)
    {
        int sum = nums[i - 2] + nums[i - 1] + nums[i];
        if (sum > prev)
        {
            ++count;
        }
        prev = sum;
    }

    cout << count << '\n';
}

File: 2021/01/python/pt2.py
import os
with open(f'{os.path.dirname(__file__)}/../input.txt', 'r') as file:
    puzzle_input = [int(i) for i in file.read().splitlines()]

count = prev_sum = 0
for idx, i in enumerate(puzzle_input):
    if idx >= 2:
        sum = i + puzzle_input[idx - 1] + puzzle_input[idx - 2]
        if sum > prev_sum:
            count += 1
        prev_sum = sum

print(count-1)
